internal/utils/convert_types.go

```diff
@@ -0,0 +1,26 @@
+package utils
+
+import (
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+func ConvertNamespacedNameToStrings(toBeConverted []types.NamespacedName) []string {
```

Comment:
This function isn't converting `[]NamespacedName` to `[]string` but `[]NamespacedName` to a very specific KMM label list.

How about we just return lists from `removeOrphanedLabels` and `addEqualLabels` instead of lists of `NamespacedName` and then write another logic to convert it?

We can completely remove this file IMO.

Reply:
@ybettan so in `removeOrphanedLabels` and `addEqualLabels` we will return []string and we should have a function that will convert the []string label list to a []type.NamespaceName list?

Reply:
No. I mean instead of:
* Returning `[]NamespacedName` and then write code to convert it

Let's just:
* Return `[]string` (then we don't need any logic to convert it) - the only use we do with `[]NamespacedName` is to convert it so it is just redundant.

Reply:
The file was deleted